### PR TITLE
364244: Remove Private Endpoints in SND3 Storage Accounts in DEFRA Dev Tenant

### PR DESCRIPTION
**Pull request by santosh-gh**
[AB#364244](https://dev.azure.com/defragovuk/93c65135-d16b-49d6-a191-4a5313532779/_workitems/edit/364244)

# **What this PR does / why we need it**:
It will remove/restrict to create private end points for the storage accounts in SND environments(DefraDev Tenant).
No manual step required for adding the private IP to the DNS Zone and will not block teams using SND environments 

# **Special notes for your reviewer**
A flag as "enablePrivateEndpoint" is passed from the patch.yaml of each environment (snd,dev, tst, pre and prd).
For SND environment the flag enablePrivateEndpoint is false 
For other environments (dev, tst, pre and prd) enablePrivateEndpoint is true

This flag enablePrivateEndpoint  being used in adp-aso-helm-library to conditionally create storage accounts with private endpoints.

# Testing
Any relevant testing information and pipeline runs.
https://dev.azure.com/defragovuk/DEFRA-FFC/_build/results?buildId=541817&view=results
https://dev.azure.com/defragovuk/DEFRA-FFC/_build/results?buildId=541491&view=results

# Checklist (please delete before completing or setting auto-complete)
- [x] Story Work items associated (not Tasks)
- [x] Successful testing run(s) link provided
- [x] Title pattern should be `{work item number}: {title}`
- [x] Description covers all the changes in the PR
- [ ] This PR contains documentation
- [ ] This PR contains tests

# **How does this PR make you feel**:

### CODE REVIEW

File: services/coreai/coreai-mcu/core-ai-techradar/infra/dev/01/patch.yaml [hunk 1]
```diff
@@ -43,3 +43,4 @@ spec:
       kubernetes_cluster: ${CLUSTER_NAME}
       kubernetes_namespace: ${TEAM_NAMESPACE}
       kubernetes_label_serviceCode: ${SERVICE_CODE}
+    enablePrivateEndpoint: "true"
```
Comment (DaRoszko) [inline services/coreai/coreai-mcu/core-ai-techradar/infra/dev/01/patch.yaml:46]: format is off? indentation? also does this need to be in scaffolder/api?

Comment (ajaysoni-cap) [reply services/coreai/coreai-mcu/core-ai-techradar/infra/dev/01/patch.yaml:46]: I'll add it to API templates... I think indentation looks right to me.

Comment (DaRoszko) [reply services/coreai/coreai-mcu/core-ai-techradar/infra/dev/01/patch.yaml:46]: Should the PE thing be hard coded? or is there any logic in the HELM that makes it dynamic.. or is it hard coded only?

Comment (kbabigumira) [reply services/coreai/coreai-mcu/core-ai-techradar/infra/dev/01/patch.yaml:46]: @santosh-gh  as discussed, let us set the values of true/false at the environment level and reference the variable instead.

Comment (santosh-gh) [reply services/coreai/coreai-mcu/core-ai-techradar/infra/dev/01/patch.yaml]: The value is set in the environment level and is referred. But instated of true/false I have set enable/disable as yaml/flux does not support true/false and yes/no.

**Review by kbabigumira — APPROVED** [review]

**Comment by ppotturi** [comment]
@santosh-gh Templates updated?

File: services/ffc/ffc-demo/base/patch/kustomize.yaml [hunk 2]
```diff
@@ -26,6 +26,7 @@ spec:
       SERVICE_CPU_QUOTA: "3000"
```
Comment (ajaysoni-cap) [inline services/ffc/ffc-demo/base/patch/kustomize.yaml:26]: This one is also not having correct variables..

Comment (santosh-gh) [reply services/ffc/ffc-demo/base/patch/kustomize.yaml:26]: Proper variables are updated

File: services/coreai/coreai-mcu/base/patch/kustomize.yaml [hunk 3]
```diff
@@ -26,6 +26,7 @@ spec:
       SERVICE_CPU_QUOTA: "5000"
```
[the remark this answers is not in the document]
Comment (santosh-gh) [reply services/coreai/coreai-mcu/base/patch/kustomize.yaml]: variables are updated

**Review by kbabigumira — APPROVED** [review]

**Review by ganeshg2491 — APPROVED** [review]

**Review by DaRoszko — APPROVED** [review]

**Pull request merged**